cognite/neat/core/utils.py

```diff
@@ -156,11 +158,48 @@ def retry_decorator(max_retries=2, retry_delay=3, component_name=""):
     def decorator(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
+            previous_exception = None
             for attempt in range(max_retries + 1):
                 try:
                     logging.debug(f"Attempt {attempt + 1} of {max_retries + 1} for {component_name}")
                     return func(*args, **kwargs)
+                except CogniteReadTimeout as e:
+                    previous_exception = e
+                    if attempt < max_retries:
+                        logging.error(
+                            f"""CogniteReadTimeout retry attempt {attempt + 1} failed for {component_name} .
+                            Retrying in {retry_delay * 2} second(s). Error:"""
+                        )
+                        logging.error(e)
+                        time.sleep(retry_delay * 2)
+                    else:
+                        raise e
+                except CogniteDuplicatedError as e:
+                    if isinstance(previous_exception, CogniteReadTimeout):
+                        # if previous exception was a timeout, we can't be sure if the items were created or not
+                        if len(e.successful) == 0 and len(e.failed) == 0 and len(e.duplicated) >= 0:
+                            logging.warning(
+                                f"Duplicate error for {component_name} . All items already exist in CDF. Suppressing error."
+                            )
+                            return
+                        else:
+                            # can happend because of eventual consistency. Retry with delay to allow for CDF to catch up
+                            if attempt < max_retries:
+                                logging.error(
+                                    f"""CogniteDuplicatedError retry attempt {attempt + 1} failed for {component_name} .
+                                      Retrying in {retry_delay} second(s). Error:"""
+                                )
+                                logging.error(e)
+                                # incerasing delay to allow for CDF to catch up
+                                time.sleep(retry_delay * 2)
+                            else:
+                                raise e
```

Comment:
This will not work I think. If you get a duplicated error, you need to look into the error for the details on which resource was triggering it and which do not exists. There is no guarantee that all resources exist when you get a DuplicationError.

Here is an example on how to do a new call with only the timeseries that does not exists
```python
    try:
        client.time_series.create(time_series_list)
    except CogniteDuplicatedError as e:
        exists = {d["externalId"] for d in e.duplicated}
        missing_time_series = [t for t in time_series_list if t.external_id not in exists]
        client.time_series.create(missing_time_series)
```